addons/account/tests/test_account_move_reconcile.py

```diff
@@ -5187,3 +5187,65 @@ def test_cash_basis_with_analytic_distribution_analytic_tax(self):
             },
         ]
         self.assertRecordValues(caba_move.line_ids.sorted('id').sorted('sequence'), expected_caba_move_line_values)
+
+    def test_partial_payments_auto_validation(self):
+        def create_move_payment(move, payment_amount, with_outstanding_account=False):
+            payment = self.init_payment(amount=payment_amount, post=False, currency=move.currency_id)
+            payment.payment_method_line_id = self.company_data['default_journal_bank'].inbound_payment_method_line_ids[0 if with_outstanding_account else 1]
```

Comment:
How do you know which one has the outstanding?
Maybe use a `filtered_domain`
```python
payment.payment_method_line_id = self.company_data['default_journal_bank'].inbound_payment_method_line_ids.filtered_domain([
    ('payment_account_id', '!=' if with_outstanding_account else "=", False),
])
```
(or the equivalent with `filtered`)